### DISCUSSION TITLE
Line wrapping in HTML-like files

### DISCUSSION THREAD

**Discussion by damithc** [TEAMMATES/teammates#11108] [General]
I guess we are not using line wrapping in HTML-like files. IIRC, we had good reasons for doing so (e.g., to reduce 'blast radius' when updating the content later). Line length is not normally a problem when using an IDE as IDEs can do soft-wrapping of lines.

But it does makes PR reviews a pain. e.g., https://github.com/TEAMMATES/teammates/pull/11049 Try reviewing instructor-help-page.module.ts and you'll see what I mean. 
e.g., scroll right to read the line, then scroll down to find the horizontal scroll bar, then scroll left, then scroll up to find the location where I need to click the `+` button, and now I can't find which line I was reading in the first place!

**Comment by wkurniawan07** [comment]
Blast radius is the reason for deprecating custom indentation rule, not line-wrapping HTML or any other language. Line-wrapping is generally favored for any language except when circumstances do not allow (e.g. hyperlinks cannot be line-wrapped no matter how long).

**Reply by damithc** [reply]
The _blast radius_ problem can affect paragraph text content, which can appear in HTML files (and also other text files such as *.md files).

Consider this:
```
<body>
    A flower, sometimes known as a bloom or blossom, is the reproductive structure found in
    flowering plants (plants of the division Magnoliophyta, also called angiosperms). The
    biological function of a flower is to facilitate reproduction, usually by providing a mechanism
    for the union of sperm with eggs. Flowers may facilitate outcrossing (fusion of sperm and
    eggs from different individuals in a population) resulting from cross pollination or allow
    selfing (fusion of sperm and egg from the same flower) when self pollination occurs.
</body>
```
Adding a few words to the first sentence will require every line of the paragraph to be adjusted for length, increasing the blast radius unnecessarily.

A workaround could be to also line break at sentence boundaries, in addition to within sentences.
```
<body>
    A flower, sometimes known as a bloom or blossom, is the reproductive structure found in
    flowering plants (plants of the division Magnoliophyta, also called angiosperms).
    The biological function of a flower is to facilitate reproduction, usually by providing a mechanism
    for the union of sperm with eggs.
    Flowers may facilitate outcrossing (fusion of sperm and eggs from different individuals
    in a population) resulting from cross pollination or allow selfing (fusion of sperm and egg
    from the same flower) when self pollination occurs.
</body>
```
Now, only the first two lines need to change when updating the first sentence.
As HTML ignores line breaks anyway, the output will still appear as a paragraph.

**Reply by wkurniawan07** [reply]
> [Adding a few words to the first sentence] [will require every line of the paragraph to be adjusted for length]

The second part is not a logical conclusion to the first part unless we make it a rule, i.e. we create our own problem of line length restriction. It is not a problem due to the text nature of HTML. Consider:

```java
String abc = something + somethingElse0 + somethingElse1 + somethingElse2
        + somethingElse3 + somethingElse4 + somethingElse5 + somethingElse6
        + somethingElse7 + somethingElse8 + somethingElse9 + somethingElse10;
```

If we add `somethingElse1_1` between `somethingElse1` and `somethingElse2`, the same problem will happen.
